freud/msd.pyx

```diff
@@ -142,19 +142,15 @@ cdef class MSD:
                 positions are assumed to be unwrapped already.
         """  # noqa: E501
 
-        positions = freud.common.convert_array(
-            positions, 3, dtype=np.float32, contiguous=True,
-            array_name="positions")
+        positions = freud.common.convert_array(positions, 3)
         if positions.shape[2] != 3:
             raise TypeError(
                 'positions should be a 3-dimensional array of shape'
                 '(N_frames, N_particles, 3)'
             )
 
         if images is not None:
-            images = freud.common.convert_array(
-                images, 3, dtype=np.float32, contiguous=True,
-                array_name="images")
+            images = freud.common.convert_array(images, 3)
```

Comment:
@vyasr Why do we treat `images` as floating values?

Reply:
There is no good reason for that. @jinsoo960 please change that to `np.int32`. Good catch.